File: http/src/main/kotlin/pt/jsal/achman/model/game/UpdateGameInput.kt
package pt.jsal.achman.model.game

import pt.jsal.achman.game.GameGenre
import pt.jsal.achman.game.GamePlatform
import pt.jsal.achman.game.GameSource

data class UpdateGameInput(
    val externalGameId: String?,
    val name: String?,
    val genres: List<GameGenre>?,
    val platform: GamePlatform?,
    val releaseYear: String?,
    val source: GameSource?,
    val cover: String?,
)

File: frontend/src/types/game/UpdateGameInput.ts
import {GameSource} from "./GameSource.ts";
import {GameGenre} from "./GameGenre.ts";
import {GamePlatform} from "./GamePlatform.ts";

export interface UpdateGameInput {
    externalGameId: string;
    name: string;
    genres?: GameGenre[];
    platform?: GamePlatform;
    releaseYear?: String;
    source: GameSource;
    cover?: string;
}